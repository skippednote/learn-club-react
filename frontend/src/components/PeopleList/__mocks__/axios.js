const people = [
  {
    id: 0,
    name: 'Ankur',
    role: 'CEO'
  },
  {
    id: 1,
    name: 'Aliya',
    role: 'Developer'
  },
  {
    id: 2,
    name: 'Bassam',
    role: 'Developer'
  }
];

module.exports = {
  get: jest.fn(() => Promise.resolve({ data: people }))
};
