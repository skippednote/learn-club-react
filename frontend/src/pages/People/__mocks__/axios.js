const fixture = [
  {
    id: 0,
    name: 'Ankur',
    role: 'CEO'
  },
  {
    id: 1,
    name: 'Aliya',
    role: 'Developer'
  },
  {
    id: 2,
    name: 'Bassam',
    role: 'Developer'
  }
];

module.exports = {
  fixture,
  get: jest.fn(() => Promise.resolve({ data: fixture }))
};
