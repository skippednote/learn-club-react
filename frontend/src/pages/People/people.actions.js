// @flow
import type {
  Person,
  GetPeopleAction,
  GetPeopleSuccessAction,
  GetPeopleFailureAction
} from './people.types';

export function getPeopleRequest(): GetPeopleAction {
  return {
    type: 'GET_PEOPLE',
    loading: true
  };
}

export function getPeopleSuccess(
  people: Array<Person>
): GetPeopleSuccessAction {
  return {
    type: 'GET_PEOPLE_SUCCESS',
    loading: false,
    people
  };
}

export function getPeopleFailure(error: string): GetPeopleFailureAction {
  return {
    type: 'GET_PEOPLE_FAILURE',
    loading: false,
    error
  };
}
