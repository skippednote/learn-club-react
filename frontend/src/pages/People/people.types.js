// @flow
export type Person = {
  id: number,
  name: string,
  role: string
};

export type State = {
  people: Array<Person>,
  error: string | null,
  loading: boolean
};

export type Props = {
  people: Array<Person>,
  error: string | null,
  loading: boolean,
  dispatch: any
};

export type GetPeopleAction = {
  type: 'GET_PEOPLE',
  loading: boolean
};

export type GetPeopleSuccessAction = {
  type: 'GET_PEOPLE_SUCCESS',
  loading: boolean,
  people: Array<Person>
};

export type GetPeopleFailureAction = {
  type: 'GET_PEOPLE_FAILURE',
  loading: boolean,
  error: string
};

export type Action =
  | GetPeopleAction
  | GetPeopleSuccessAction
  | GetPeopleFailureAction;

export type Dispatch = (action: Action) => void;
