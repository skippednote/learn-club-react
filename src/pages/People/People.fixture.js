export default [
  {
    id: 0,
    name: 'Ankur',
    role: 'CEO'
  },
  {
    id: 1,
    name: 'Aliya',
    role: 'Developer'
  },
  {
    id: 2,
    name: 'Bassam',
    role: 'Developer'
  }
];
