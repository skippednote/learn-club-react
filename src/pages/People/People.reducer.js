// @flow
import type { State, Action } from './People.types';

const PeopleReducer = (
  state: State = { people: [], error: null, loading: false },
  action: Action
): State => {
  switch (action.type) {
    case 'GET_PEOPLE':
      return {
        ...state,
        loading: action.loading
      };
    case 'GET_PEOPLE_SUCCESS':
      return {
        ...state,
        loading: action.loading,
        people: action.people
      };
    case 'GET_PEOPLE_FAILURE':
      return {
        ...state,
        loading: action.loading,
        error: action.error
      };
    default:
      return state;
  }
};

export default PeopleReducer;
