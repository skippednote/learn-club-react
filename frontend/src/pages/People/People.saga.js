import axios from 'axios';
import { put, takeEvery } from 'redux-saga/effects';
import { getPeopleSuccess, getPeopleFailure } from './People.actions';

export function* getPeople(action) {
  try {
    const { data: people } = yield axios.get('/people.json');
    yield put(getPeopleSuccess(people));
  } catch (error) {
    yield put(getPeopleFailure(error));
  }
}

export default function* saga() {
  yield takeEvery('GET_PEOPLE', getPeople);
}
