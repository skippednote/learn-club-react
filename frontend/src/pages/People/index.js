// @flow
import React from 'react';
import axios from 'axios';
import PeopleList from '../../components/PeopleList';

type Person = {
  id: number,
  name: string,
  role: string
};

type State = {
  people: Array<Person>,
  error: string | null,
  loading: boolean
};

class People extends React.Component<{}, State> {
  state: State = {
    people: [],
    error: null,
    loading: false
  };

  async componentDidMount() {
    this.setState({ loading: true });

    try {
      const { data: people }: { data: Array<Person> } = await axios.get(
        '/people.json'
      );
      this.setState({ people, loading: false });
    } catch (error) {
      this.setState({ error: error, loading: false });
    }
  }
  render() {
    const { people, error, loading } = this.state;

    return (
      <div>
        <PeopleList people={people} error={error} loading={loading} />
      </div>
    );
  }
}
export default People;
