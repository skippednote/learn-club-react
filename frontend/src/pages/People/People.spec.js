import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import axios from 'axios';
import People from './index';

describe('Home', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<People />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders PeopleList component', () => {
    const wrapper = shallow(<People />);
    const counter = wrapper.find('PeopleList');
    expect(counter).toHaveLength(1);
  });

  it('requests data on #componentDidMount', async () => {
    const wrapper = shallow(<People />);
    await wrapper.instance().componentDidMount();
    expect(axios.get).toHaveBeenCalled();
    expect(axios.get).toHaveBeenCalledWith('/people.json');
  });
});
