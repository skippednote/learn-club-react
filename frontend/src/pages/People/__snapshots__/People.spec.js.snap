// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div>
  <PeopleList
    error={null}
    loading={true}
    people={Array []}
  />
</div>
`;
